src/Entity/ItemId.php

```diff
@@ -29,12 +29,16 @@ public function __construct( $idSerialization ) {
 
 	private function assertValidIdFormat( $idSerialization ) {
 		if ( !is_string( $idSerialization ) ) {
-			throw new InvalidArgumentException( '$idSerialization must be a string; got ' . gettype( $idSerialization ) );
+			throw new InvalidArgumentException( '$idSerialization must be a string' );
 		}
 
 		if ( !preg_match( self::PATTERN, $idSerialization ) ) {
 			throw new InvalidArgumentException( '$idSerialization must match ' . self::PATTERN );
 		}
+
+		if ( strlen( $idSerialization ) > 10 && substr( $idSerialization, 1 ) > 2147483647 ) {
```

Comment:
I played around with multiple ways of writing this line, and everything was ugly and not easier to read than this one. The only thing that may help is adding `.0` at the end of the constant. But this doesn't really do anything. What's critical here is the test, not the implementation.